### ISSUE TITLE
Per Layer Ternarization

### ISSUE THREAD

**Issue by HahTK** [fengfu-chris/caffe-twns#5]
The boolean controlling ternarization in binary.hpp seems to affect all inner_product and conv layers.
That suggests that either all layers are either ternarized or not. Is there a way to ternarize only some layers ?

I'm thinking of creating a new layer type inner_product_twn and conv_twn and revert the original inner_product and conv layers to baseline caffe behavior. Do you see any problems with this approach ?

Also, the alpha and delta are computed separately for each layer correct ?

**Comment by fengfu-chris** [comment]
@HahTK Yes, per layer ternarization is possible when you revise the source file of inner_product layers and cnn layers. One way is to add a parameter that controls whether a layer should be ternarized. In the .prototxt file that defines the network architecture, the true value of the parameters can be defined and be passed to the source files.
